### PR TITLE
expose algorithm from JWKS on OIDC Discovery Provider

### LINKED ISSUE #2753: Feature Request: Explicitly expose algorithm from JWKS on OIDC Discovery Provider
Hi! I am trying to use SPIRE OIDC Discovery Provider to authenticate a service. When I try to validate a JWT SVID through Spire OIDC Discovery Provider, the validation service is expecting the `alg` property in the key. If it is not present, the validation fails. Thus, I can't validate a JWT SVID this way.
So I would like to request the option to explictly show the algorith from the keys, in the `alg` field. It could be a configuration which can be enabled, just like the `set_key_use`.

Thanks

### PR DESCRIPTION
**Pull request by loveyana**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**

OIDC Discovery Provider

**Description of change**

add `set_key_algo` to `oidc-discovery-provider` config, If true, the `algo` parameter on JWKs will be set to `Algorithm` for each key, \<RS256\|ES256\|ES384\>.

**Which issue this PR fixes**

fixes #2753

### CODE REVIEW

**Review by evan2645 — COMMENTED** [review]
Thank you for this @loveyana! A few minor comments, let me know what you think

File: support/oidc-discovery-provider/handler.go [hunk 1]
```diff
@@ -114,6 +119,32 @@ func (h *Handler) serveKeys(w http.ResponseWriter, r *http.Request) {
...
+				case 256:
+					alg = jose.ES256
+				case 384:
+					alg = jose.ES384
+				default:
+					// unable to determine signature algorithm for EC public key size
+				}
+			default:
+				// unable to determine signature algorithm for public key type
+			}
```
[the remark this answers is not in the document]
Comment (loveyana) [reply support/oidc-discovery-provider/handler.go]: Yep, originally there was no log variable passed in http handle, I have now added it and provided a log for `GetPublicKeyAlgorithm` failure with keyID

File: support/oidc-discovery-provider/handler.go [hunk 2]
```diff
@@ -114,6 +119,32 @@ func (h *Handler) serveKeys(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	if h.setKeyAlgo {
+		for index, k := range jwks.Keys {
```
[the remark this answers is not in the document]
Comment (loveyana) [reply support/oidc-discovery-provider/handler.go]: I added the `GetPublicKeyAlgorithm` method in cryptoutil to let jwt svid sign also use it, hoping to solve your concerns. At the same time, I added `enrichJwksKeys` to the oidc handler to handle the key enrichment requirements of jwks at the same time.

File: support/oidc-discovery-provider/README.md [hunk 3]
```diff
@@ -37,6 +37,7 @@ The configuration file is **required** by the provider. It contains
 | `domains`               | strings | required       | One or more domains the provider is being served from.                       |          |
 | `insecure_addr`         | string  | optional[3]    | Exposes the service on http.                                                 |          |
 | `set_key_use`           | bool    | optional       | If true, the `use` parameter on JWKs will be set to `sig`.                   | `false`  |
+| `set_key_algo`           | bool    | optional       | If true, the `algo` parameter on JWKs will be set to `Algorithm` for each key, \<RS256\|ES256\|ES384\>.                  | `false`  |
```
[the remark this answers is not in the document]
Comment (loveyana) [reply support/oidc-discovery-provider/README.md]: LGTM, I removed this option and made it the default configuration

**Review by evan2645 — COMMENTED** [review]
This is looking great @loveyana thank you so much for the contribution! Just a couple small comments and then I think we're ready to merge this

File: pkg/common/cryptoutil/keys.go [hunk 4]
```diff
@@ -48,3 +51,28 @@ func KeyMatches(privateKey crypto.PrivateKey, publicKey crypto.PublicKey) (bool,
 		return false, fmt.Errorf("unsupported private key type %T", privateKey)
 	}
 }
+
+func GetPublicKeyAlgorithm(publicKey interface{}) (jose.SignatureAlgorithm, error) {
```
[the remark this answers is not in the document]
Comment (loveyana) [reply pkg/common/cryptoutil/keys.go]: great idea, I changed the method name and added a basic test

**Review by evan2645 — APPROVED** [review]
thank you @loveyana!

**Pull request merged**